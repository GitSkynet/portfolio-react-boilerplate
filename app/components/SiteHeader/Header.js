import styled from 'styled-components';

const Header = styled.header`
  flex: none;
  z-index: 10;
  background-color: #465a65;
`;

export default Header;
