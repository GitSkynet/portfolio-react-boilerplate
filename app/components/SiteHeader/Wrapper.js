import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  border-bottom: 0.5vh solid #465a65;
`;

export default Wrapper;
