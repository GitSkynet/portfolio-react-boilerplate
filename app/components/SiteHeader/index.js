import React from 'react';

import Header from './Header';
import Wrapper from './Wrapper';

import Brand from '../Brand';
import Widgets from '../Widgets';
import SiteNavigationBar from '../../containers/SiteNavigationBar';

const SiteHeader = () => (
  <Header role="banner">
    <Wrapper>
      <Brand />
      <Widgets />
    </Wrapper>
    <SiteNavigationBar />
  </Header>
);

export default SiteHeader;
