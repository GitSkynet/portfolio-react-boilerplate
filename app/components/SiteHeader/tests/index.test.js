import React from 'react';
import { shallow } from 'enzyme';

import SiteHeader from '../index';

describe('SiteHeader', () => {
  it('should render', () => {
    const setSiteNavIsFixedOffset = jest.fn();
    const wrapper = shallow(
      <SiteHeader
        setSiteNavIsFixedOffset={setSiteNavIsFixedOffset}
        siteNavIsAtScreenTop={false}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
