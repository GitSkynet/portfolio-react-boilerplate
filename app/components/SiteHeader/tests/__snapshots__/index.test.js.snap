// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Header__Header
    role="banner"
>
    <Wrapper__Wrapper>
        <Brand
            href="/"
            imageSource="IMAGE_MOCK"
            text="JOHAN MEESTER"
        />
        <SocialLinks />
    </Wrapper__Wrapper>
    <Connect(SiteNavigationBar) />
</Header__Header>,
  "nodes": Array [
    <Header__Header
      role="banner"
>
      <Wrapper__Wrapper>
            <Brand
                  href="/"
                  imageSource="IMAGE_MOCK"
                  text="JOHAN MEESTER"
            />
            <SocialLinks />
      </Wrapper__Wrapper>
      <Connect(SiteNavigationBar) />
</Header__Header>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SiteHeader
        setSiteNavIsFixedOffset={[Function]}
        siteNavIsAtScreenTop={false}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "setSiteNavIsFixedOffset": [Function],
          "siteNavIsAtScreenTop": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Header__Header
          role="banner"
>
          <Wrapper__Wrapper>
                    <Brand
                              href="/"
                              imageSource="IMAGE_MOCK"
                              text="JOHAN MEESTER"
                    />
                    <SocialLinks />
          </Wrapper__Wrapper>
          <Connect(SiteNavigationBar) />
</Header__Header>,
        "_debugID": 2,
        "_renderedOutput": <Header__Header
          role="banner"
>
          <Wrapper__Wrapper>
                    <Brand
                              href="/"
                              imageSource="IMAGE_MOCK"
                              text="JOHAN MEESTER"
                    />
                    <SocialLinks />
          </Wrapper__Wrapper>
          <Connect(SiteNavigationBar) />
</Header__Header>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SiteHeader
    setSiteNavIsFixedOffset={[Function]}
    siteNavIsAtScreenTop={false}
/>,
}
`;
